src/fsharp/fsi/fsi.fs

```diff
@@ -2000,31 +2064,39 @@ let internal SpawnInteractiveServer
              Server.Shared.FSharpInteractiveServer.StartServer(fsiOptions.FsiServerName,server)
          with e ->
              fprintfn fsiConsoleOutput.Error "%s" (FSIstrings.SR.fsiExceptionRaisedStartingServer(e.ToString())))
-  
+#endif
 
-let internal StartStdinReadAndProcessThread
-                                  (lcid, istateRef, errorLogger, 
-                                   fsiConsoleInput: FsiConsoleInput, 
-                                   fsiConsoleOutput: FsiConsoleOutput,
-                                   fsiStdinLexerProvider: FsiStdinLexerProvider, 
-                                   fsiInteractionProcessor : FsiInteractionProcessor,
-                                   exitViaKillThread) = 
+let internal StartStdinReadAndProcessThread(
+#if FX_LCIDFROMCODEPAGE
+                                            lcid,
+#endif
+                                            istateRef, 
+                                            errorLogger, 
+                                            fsiConsoleInput: FsiConsoleInput, 
+                                            fsiConsoleOutput: FsiConsoleOutput,
+                                            fsiStdinLexerProvider: FsiStdinLexerProvider, 
+                                            fsiInteractionProcessor: FsiInteractionProcessor,
+                                            exitViaKillThread) = 
     if !progress then fprintfn fsiConsoleOutput.Out "creating stdinReaderThread";
     let cont = ref Completed 
     let tokenizerRef = ref (fsiStdinLexerProvider.CreateStdinLexer())
+#if FX_LCIDFROMCODEPAGE
```

Comment:
Well for a start the full desktop can run the coreclr version of FSC and FS, so long as the right dependencies are available.  It's how I do debugging today.  So I am pretty certain that with a single compiler we can satisfy almost all existing scripts and type providers.  I say almost all, because there may be some bugs or scripts doing stuff that is absolutely not supported.

Our compiled code has to be able to run on both coreclr and the desktop, but for desktop only scripts and typeproviders they should in theory work fine when the compiler or FSI works against the desktop CLR.

Anyway, the above is a "in my head" goal, not necessarily a certainty.  We may well ship two compilers and FSI's, until we can ensure that transition is seamless.  Until we know the answer no code is going to go away, instead it will be #ifdef'd out.